lib/fog/core/errors.rb

```diff
@@ -11,8 +11,6 @@ def self.slurp(error, message = nil)
       end
     end
 
-    class MockNotImplemented < Fog::Errors::Error; end
```

Comment:
I'm still not sure if this exception should be extracted as well. Even that is being used only by Mock it is still an exception and split it might make it harder to figure out stuff later. What do you think?